File: contest/abc325/b.rb
n = gets.chomp.to_i
arr = []
n.times do
  w, x = gets.chomp.split(" ").map(&:to_i)
  arr << [w, x]
end

results = []
(0..23).each do |i|
  sum = 0
  arr.each do |a|
    if (i + a[1]) % 24 <= 17 && (i + a[1]) % 24 >= 9
      sum += a[0]
    end
  end
  results << sum
end

puts results.max

File: contest/abc325/b.exs
defmodule Main do
  def main do
    n =
      IO.read(:line)
      |> String.trim
      |> String.to_integer

    arr = for _i <- 0..(n-1) do
      IO.read(:line)
        |> String.trim
        |> String.split(" ")
        |> Enum.map(&String.to_integer/1)
        |> List.to_tuple
    end

    results = Enum.map(0..23, fn i ->
      Enum.reduce(arr, 0, fn {w, x}, acc ->
        if rem(i+x, 24) <= 17 and rem(i+x, 24) >= 9 do
          acc + w
        else
          acc
        end
      end)
    end)

    IO.puts(Enum.max(results))
  end
end

Main.main()
